src/views/ApartmentApp/index.js

```diff
@@ -1,96 +1,105 @@
-import React, { Component } from 'react';
+//Main apartment application page
+import React, { useState, useEffect } from 'react';
 import { Grid, Card, CardContent, Button } from '@material-ui/core/';
 import GordonLoader from '../../components/Loader';
 import StudentApplication from './components/StudentApplication';
 import StaffMenu from './components/StaffMenu';
 import user from '../../services/user';
-// import housing from '../../services/housing';
+import housing from '../../services/housing';
 import './apartmentApp.scss';
 
-export default class ApartApp extends Component {
-  constructor(props) {
-    super(props);
-    this.peopleSearch = React.createRef();
-    this.state = {
-      isStu: Boolean,
-      isHousingStaff: Boolean,
-      loading: true,
-      network: 'online',
-      userProfile: {},
-    };
-  }
+const ApartApp = (props) => {
+  const [loading, setLoading] = useState(true);
+  const [network, setNetwork] = useState('online');
+  const [userProfile, setUserProfile] = useState({});
+  const [canUseStaff, setCanUseStaff] = useState(false);
+  const [isUserStudent, setIsUserStudent] = useState(false);
 
-  componentDidMount() {
-    this.loadProfile();
-  }
+  useEffect(() => {
+    loadUserProfile();
+    checkHousingStaff();
+  });
 
   /**
    * Loads the user's profile info only once (at start)
    */
-  async loadProfile() {
-    this.setState({ loading: true });
+  const loadUserProfile = async () => {
+    setLoading(true);
     try {
-      const profile = await user.getProfileInfo();
-      this.setState({ userProfile: profile });
-      this.setState({ isStu: String(profile.PersonType).includes('stu') });
-      this.setState({ loading: false });
+      user.getProfileInfo().then((data) => {
+        setUserProfile(data);
+        data.PersonType.includes('stu') ? setIsUserStudent(true) : setIsUserStudent(false);
+      });
     } catch (error) {
```

Comment:
Using `promise().then(...)` syntax makes the Try Catch block meaningless sadly. It's probably best to switch it back to await since it's an async function.